webpack/assets/javascripts/react_app/components/HostDetails/index.js

```diff
@@ -1,155 +1,46 @@
-            </BreadcrumbItem>
-          </Breadcrumb>
-          {/* TODO: Replace all br with css */}
-          <br />
-          <br />
-          <Grid>
-            <GridItem span={3}>
-              <Title headingLevel="h5" size="2xl">
-                {/* TODO: Make a generic Skeleton HOC (withSkeleton) */}
-                {response.name || <Skeleton />}{' '}
-                <HostGlobalStatus hostName={id} />
-              </Title>
-            </GridItem>
-            <GridItem
-              style={{ marginTop: '5px', marginRight: '30px' }}
-              span={7}
-            >
-              <Badge key={1}>{response.operatingsystem_name}</Badge>{' '}
-              <Badge key={21}>{response.architecture_name}</Badge>
-            </GridItem>
-            <GridItem span={2}>
-              <ActionsBar hostName={response.name} />
-            </GridItem>
-          </Grid>
-          <SkeletonLoader
-            skeletonProps={{ width: 400 }}
-            status={status || STATUS.PENDING}
-          >
-            {response && (
-              <Text component={TextVariants.span}>
-                <RelativeDateTime date={response.created_at} defaultValue="N/A">
-                  {date =>
-                    sprintf(__('Created %s by %s'), date, response.owner_name)
-                  }
-                </RelativeDateTime>{' '}
-                <RelativeDateTime date={response.updated_at} defaultValue="N/A">
-                  {date => sprintf(__('(updated %s)'), date)}
-                </RelativeDateTime>
-              </Text>
-            )}
-          </SkeletonLoader>
-          <br />
-        </div>
-        {tabs && (
-          <TabRouter
-            response={response}
-            hostName={id}
-            status={status}
-            tabs={tabs}
-            router={history}
-          >
-            <Tabs
-              style={{
-                width: window.innerWidth - (isNavCollapsed ? 95 : 220),
-              }}
-              activeKey={hash.slice(2).split('/')[0]}
-            >
-              {tabs.map(tab => (
-                <Tab key={tab} eventKey={tab} title={tab} href={`#/${tab}`} />
-              ))}
-            </Tabs>
-          </TabRouter>
-        )}
-      </PageSection>
-    </>
-  );
+  LEGACY_DETAILS_PATH,
+  RENDER_LEGACY,
+  RENDER_REACT,
+  TAB_PREFIX,
+} from './consts';
+
+const HostUISwitcher = props => {
+  const { hostDetailsUI } = useForemanSettings();
+  const {
+    match: {
+      params: { id },
+    },
+    location: { state, hash },
+  } = props;
+
+  if (hash.startsWith(TAB_PREFIX)) return <HostDetailsPage {...props} />;
+  switch (state?.render) {
+    case RENDER_LEGACY:
+      return <Redirect to={`${LEGACY_DETAILS_PATH}/${id}`} />;
+    case RENDER_REACT:
+      return <HostDetailsPage {...props} />;
+    default:
+      return hostDetailsUI ? (
+        <HostDetailsPage {...props} />
+      ) : (
+        <Redirect to={`${LEGACY_DETAILS_PATH}/${id}`} />
+      );
```

Comment:
@ezr-ondrej, this allows consumers to switch programmatically between react / legacy page.
switchers buttons will be added in a separate PR